kratos/geometries/quadrilateral_3d_4.h

```diff
@@ -649,16 +640,17 @@ template<class TPointType> class Quadrilateral3D4
      * @see DeterminantOfJacobian
      * @see InverseOfJacobian
      */
-    JacobiansType& Jacobian( JacobiansType& rResult,
-                                     IntegrationMethod ThisMethod ) const override
+    JacobiansType& Jacobian( 
+        JacobiansType& rResult,
+        IntegrationMethod ThisMethod 
+        ) const override
     {
-        //getting derivatives of shape functions
+        // Getting derivatives of shape functions
         ShapeFunctionsGradientsType shape_functions_gradients =
-            CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
-        //getting values of shape functions
+        CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
+        // Getting values of shape functions
         Matrix shape_functions_values =
-            CalculateShapeFunctionsIntegrationPointsValues( ThisMethod );
-        //workaround by riccardo...
+        CalculateShapeFunctionsIntegrationPointsValues( ThisMethod );

```

Comment:
Avoid Matrix generation by calling the ShapeFunctionsValues method

Reply:
Ok